packages/cli-repl/src/startup-timing.ts

```diff
@@ -0,0 +1,50 @@
+const jsTimingEntries: [string, bigint][] = [];
+const timing: {
+  markTime: (label: string) => void;
+  getTimingData: () => [string, number][];
+} = !process.env.MONGOSH_SHOW_TIMING_DATA
+  ? {
+      markTime: () => {
+        /* ignore */
+      },
+      getTimingData: () => [],
+    }
+  : (process as any).boxednode
+  ? {
+      markTime: (process as any).boxednode.markTime,
+      getTimingData: (process as any).boxednode.getTimingData,
+    }
+  : {
+      markTime: (label: string) => {
+        jsTimingEntries.push([label, process.hrtime.bigint()]);
+      },
+      getTimingData: () => {
+        const data = jsTimingEntries.sort((a, b) => Number(a[1] - b[1]));
+        // Adjust times so that process initialization happens at time 0
+        return data.map(([label, time]) => [label, Number(time - data[0][1])]);
+      },
+    };
+
+export const markTime = timing.markTime;
+export const getTimingData = timing.getTimingData;
+
+if (process.env.MONGOSH_SHOW_TIMING_DATA) {
+  process.on('exit', function () {
+    const rawTimingData = getTimingData();
+    if (process.env.MONGOSH_SHOW_TIMING_DATA === 'json') {
+      console.log(JSON.stringify(rawTimingData));
```

Comment:
Not sure if this makes sense: as this measures startup time, not exit time, maybe we can always write this to log file on exit? Might be helpful for debugging issues without asking issue reporters to redo whatever they were doing, but with these env vars passed if they already provided a log file

Reply:
We can obviously adjust this a bit if we want, I think there was also a suggestion to include this in telemetry, I don’t think we actually created a ticket or anything for that though so I’ll ping Gaurab about it